source/extensions/common/wasm/wavm/wavm.cc

```diff
@@ -585,6 +620,36 @@ template void getFunctionWavm<uint32_t, uint32_t, uint32_t, uint32_t, uint32_t,
     std::function<uint32_t(Context*, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t,
                            uint32_t, uint32_t, uint32_t, uint32_t)>*);
 
+template <typename T> T getValue(IR::Value) {}
+template <> int32_t getValue(IR::Value v) { return v.i32; }
+template <> uint32_t getValue(IR::Value v) { return v.u32; }
+template <> int64_t getValue(IR::Value v) { return v.i64; }
+template <> uint64_t getValue(IR::Value v) { return v.u64; }
+template <> float getValue(IR::Value v) { return v.f32; }
+template <> double getValue(IR::Value v) { return v.f64; }
+
+template <typename T> T WavmGlobal<T>::get() {
+  return getValue<T>(getGlobalValue(wavm_->context_, global_));
```

Comment:
This implementation remind me that the return type should be [std::decay_t<T>](https://en.cppreference.com/w/cpp/types/decay) to avoid get reference (e.g, T = int32_t&)

Reply:
discussed offline.  This is only used for templates explicitly instantiated in this file, so there is no chance of accidental &.